examples/bind-shopping.amp.html

```diff
@@ -0,0 +1,454 @@
+    }
+    .thumbnails amp-img[selected] {
+      border: solid 2px gray;
+    }
+    .thumbnails amp-img[out-of-stock] {
+      background-color: red;
+    }
+    amp-accordion {
+      margin-top: 10px;
+    }
+    .sizes button {
+      display: block;
+      width: 100%;
+      border: solid 1px gray;
+      padding: 5px;
+      background-color: #eee;
+    }
+    .change-size {
+      display: block;
+      width: 200px;
+      height: 30px;
+      padding: 5px;
+      background-color: #666;
+      color: white;
+    }
+    button[hidden] {
+      display: none;
+    }
+    p[hidden] {
+      display:none;
+    }
+  </style>
+</head>
+
+<body>
+  <h1 class="header">AMPWEAR</h1>
+  <div class="container">
+    <h2>MEN'S COTTON SHORT SLEEVE T-SHIRT</h2>
+
+    <amp-bind-dataset layout=nodisplay>
+      <script type="application/json">
+        {
+          "items": [
+            {
+              "id": "0",
+              "name": "09 BLACK",
+              "image": "./shirts/black.jpeg",
+              "sizes": ["S", "L"]
+            },
+            {
+              "id": "1",
+              "name": "34 BROWN",
+              "image": "./shirts/brown.jpeg",
+              "sizes": ["M", "L"]
+            },
+            {
+              "id": "2",
+              "name": "03 GRAY",
+              "image": "./shirts/gray.jpeg",
+              "sizes": ["S"]
+            },
+            {
+              "id": "3",
+              "name": "69 NAVY",
+              "image": "./shirts/navy.jpeg",
+              "sizes": ["S", "L"]
+            },
+            {
+              "id": "4",
+              "name": "19 WINE",
+              "image": "./shirts/wine.jpeg",
+              "sizes": []
+            },
+            {
+              "id": "5",
+              "name": "61 BLUE",
+              "image": "./shirts/blue.jpeg",
+              "sizes": ["L"]
+            },
+            {
+              "id": "6",
+              "name": "59 DARK GREEN",
+              "image": "./shirts/dark-green.jpeg",
+              "sizes": ["S"]
+            },
+            {
+              "id": "7",
+              "name": "02 LIGHT GRAY",
+              "image": "./shirts/light-gray.jpeg",
+              "sizes": ["S", "M", "L"]
+            },
+            {
+              "id": "8",
+              "name": "00 WHITE",
+              "image": "./shirts/white.jpeg",
+              "sizes": []
+            }
+          ]
+        }
+      </script>
+    </amp-bind-dataset>
+
+    <!--
+    <amp-carousel slide=0 id="carousel-1" width=1 height=1 layout=responsive type=slides controls
+        bind-attr="{slide: selected}"
+        on="slidechange:setState({selected: ampEventData})">
+    -->
+    <amp-carousel slide=0 id="carousel-1" width=1 height=1 layout=responsive type=slides controls>
+      <amp-bind-attribute layout=nodisplay
+          attr="slide"
+          value="{{selected}}">
+      </amp-bind-attribute>
+      <amp-bind-event layout=nodisplay
```

Comment:
What is `amp-bind-event`? Why is it needed? How is it different from just `amp-bind` above? Generally, to me, presence of events immediately communicates two-way binding which we are here trying to avoid.